src/server/services/mongo-service.ts

```diff
@@ -1,48 +1,25 @@
 "use server";
 
-import RecordedGameModel, {
-  IRecordedGame,
-} from "@/db/mongo/model/RecordedGameModel";
+import RecordedGameModel from "@/db/mongo/model/RecordedGameModel";
 import getMongoClient from "@/db/mongo/mongo-client";
-import { MythRec } from "@/types/MythRecs";
+import { IRecordedGame } from "@/types/RecordedGame";
+import { removeMongoObjectID } from "@/utils/utils";
 
-export async function queryMythRecs(pageIndex: number): Promise<MythRec[]> {
+export async function queryMythRecs(pageIndex: number): Promise<IRecordedGame[]> {
   console.log("pageIndex", pageIndex);
   const PAGE_SIZE = 2
   const offset = (pageIndex * PAGE_SIZE)
   await getMongoClient();
   try {
-    const result: IRecordedGame[] = await RecordedGameModel.find()
+    const result = await RecordedGameModel.find().lean()
       .sort({ createdAt: -1 })
       .skip(offset)
       .limit(PAGE_SIZE)
 
-    const mythRecs: MythRec[] = result.map((video) => {
-      const mappedPlayerData = video.playerdata.map((player) => {
-        return {
-          name: player.name,
-          team: player.teamid,
-          civ: player.civ,
-          civList: player.civlist,
-          rating: player.rating,
-          rank: player.rank,
-          powerRating: player.powerrating,
-          winRatio: player.winratio,
-          civWasRandom: player.civwasrandom,
-          color: player.color,
-        };
-      });
-      return {
-        gameGuid: video.gameguid,
-        playerData: mappedPlayerData,
-        mapName: video.gamemapname,
-        createdAt: video.createdAt,
-        uploadedBy: video.uploadedBy ?? "FitzBro", // remove these defaults later
-        gameTitle: video.gameTitle ?? "Retold Rec", // remove these defaults later
-        downloadCount: video.downloadCount ?? 0,
-      };
-    });
-    return mythRecs;
+    // Remove _id from each playerData object and the object as a whole
+    result.map((rec) => { removeMongoObjectID(rec.playerData)});
```

Comment:
Is the _objectId being removed so that it matches the model? Alternatively, you can use a Mongo aggregate pipeline and project the fields you want to return. Might work better.

Reply:
This relates to that call stack size exceeded error in String.replace that you had when trying to return the whole records (which is why `MythRec` and `IPlayerData` came about originally). It turns out that using `lean()` and returning the query result array unmodified makes the error go away, but instead it writes quite a lot of warnings instead: these warnings are all because the function is returning an object properties that nonserialisable, so Next/React complain when trying to send it to the client. Something about the non-lean output presumably is causing too much recursion when trying to generate the warning, giving the strange error.

After using `lean()` there is only one nonserialisable property left: mongoose's `_id` field that it seemingly tries to add to everything. So removing this field from the returned data (it's also added to each member of `playerData`) before passing it to the client gets rid of all the warnings. I don't know if there's a better/tidier way to do this, though.

Reply:
Let's use .lean() and then .map the results.